### PR TITLE
dependency checklist and flow support

### PR DESCRIPTION
**Pull request by damfinkel**
## Summary

- Added checklist for some dependency groups
- Added precommit hook that runs the linter before committing
- Added flow support

## Screenshot

screen shot 2018-07-19 at 14 11 04

### CODE REVIEW

File: generators/app/templates/flowconfig [hunk 1]
```diff
@@ -0,0 +1,12 @@
+[ignore]
+
+[include]
+
+[libs]
+
+[lints]
+
+[options]
+emoji=true
```
Comment (mvbattan) [inline generators/app/templates/flowconfig:10]: ✨

File: generators/app/templates/eslintrc.js [hunk 2]
```diff
@@ -1,5 +1,6 @@
 module.exports = {
-  extends: 'react-app',
+  extends: ['wolox'<% if (flow) { %>, 'plugin:flowtype/recommended'<% } %>],  <% if (flow) { %>
+  plugins: ['flowtype'],<% } %>
```
Comment (mvbattan) [inline generators/app/templates/eslintrc.js:3]: 🎉

File: .eslintignore [hunk 3]
```diff
@@ -0,0 +1 @@
+/generators/app/templates/eslintrc.js
```
Comment (mvbattan) [inline .eslintignore]: Missing EOF

**Review by mvbattan — APPROVED** [review]

File: generators/app/tasks/installDependencies.js [hunk 4]
```diff
@@ -1,27 +1,19 @@
...
-  'react-redux',
   'react-router',
-  'react-router-dom',
-  'react-router-redux',
-  'redux',
-  'redux-beacon',
-  'redux-form',
-  'redux-thunk',
-  'seamless-immutable'
+  'react-router-dom'
```
Comment (FrankIglesias) [inline generators/app/tasks/installDependencies.js:16]: I'd add husky if as a dependency

Comment (damfinkel) [reply generators/app/tasks/installDependencies.js:16]: husky is already added as a non-optional devDependency

File: package.json [hunk 5]
```diff
@@ -40,14 +41,15 @@
...
-    "prettier-eslint": "^6.4.2",
     "eslint-config-airbnb": "^15.1.0",
     "eslint-config-prettier": "^2.3.0",
     "eslint-plugin-flowtype": "^2.30.3",
     "eslint-plugin-import": "^2.7.0",
     "eslint-plugin-jsx-a11y": "^5.1.1",
     "eslint-plugin-prettier": "^2.0.1",
-    "eslint-plugin-react": "^7.1.0"
+    "eslint-plugin-react": "^7.1.0",
+    "husky": "^0.14.3",
```
Comment (FrankIglesias) [inline package.json:51]: this will add also to the generated project? I thought this was for the bootstrap itself

Comment (damfinkel) [reply package.json:51]: Both. I wanted to add husky to have a precommit push in the react-bootstrap project so we avoid pushing with linter errors, and also I wanted to add husky as a dev dependency in the generated project

Comment (FrankIglesias) [reply package.json:51]: greatt!!!

**Pull request merged**